File: strings/length-of-last-word/length-of-last-word.ts
function lengthOfLastWord(s: string): number {
  let len = 0;

  for (let i = s.length - 1; i >= 0; i--) {
    if (s[i] != ' ') {
      len++;
    } else if (len != 0) {
      return len;
    }
  }
  return len;
}

File: strings/length-of-last-word/length-of-last-word.go
package lengthoflastword

func lengthOfLastWord(s string) int {
	l := 0

	for i := len(s) - 1; i >= 0; i-- {
		if s[i] != ' ' {
			l++
		} else if l != 0 {
			return l
		}
	}
	return l
}
